pkg/store/store.go

```diff
@@ -296,30 +300,59 @@ func (t *Store) Count(prefix schema.KeyPrefix) (count *schema.ItemsCount, err er
 	return
 }
 
-// fixme(leogr): need to be optmized
-func (t *Store) itemAt(readTs uint64) (version uint64, key, value []byte, err error) {
-	stream := t.db.NewStreamAt(readTs)
-	stream.ChooseKey = func(item *badger.Item) bool {
-		return item.UserMeta() != bitTreeEntry && item.Version() == readTs
+func (t *Store) itemAt(readTs uint64) (index uint64, key, value []byte, err error) {
+	if readTs > t.tree.w {
```

Comment:
This seems inconsistent to me. 
`t.tree.w` represents the width of the computed Merkel tree, not the last valid item index. For a short time frame, newly inserted items can be already stored in the DB although their node in the Merkel tree have not been computed yet.

Anyhow, if you used `t.tree.w` I believe `t.tree` should be read-locked at least.

Reply:
I remove the guard.
Method return all items currently stored. When the iterator miss the first element a not found error is returned.